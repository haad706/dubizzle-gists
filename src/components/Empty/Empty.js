import styled from "styled-components";
import PropTypes from "prop-types";
import EmptyContainerImage from "assets/no-data.svg";

export const Empty = ({ text }) => {
  return (
    <Container>
      <img src={EmptyContainerImage} alt="no-gists-found" />
      <h3>{text}</h3>
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 20vh;
  img {
    width: 170px;
  }
`;

Empty.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Empty;
