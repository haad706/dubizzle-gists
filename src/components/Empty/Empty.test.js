import { render } from "@testing-library/react";
import Empty from "./Empty";

describe("Empty", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<Empty text="No data found" />);
    expect(baseElement).toBeTruthy();
  });
  it("should match snapshot", () => {
    const { asFragment } = render(<Empty text="No data found" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
