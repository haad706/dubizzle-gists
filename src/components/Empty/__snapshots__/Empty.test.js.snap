// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bgqQcB jpmheq"
  >
    <img
      alt="no-gists-found"
      src="no-data.svg"
    />
    <h3>
      No data found
    </h3>
  </div>
</DocumentFragment>
`;
